website/content/en/docs/contribution-guidelines/testing.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Testing Guide
+linkTitle: Testing
+weight: 10
+---
+
+On all PRs, a suite of static and cluster tests is run against your changes in a CI environment.
+These tests can also be run locally, which is discussed [below](#local-test-environment).
+
+Static tests consist of [unit][unit-tests], formatting, and doc link tests.
+
+Cluster tests consist of several test types:
+- End-to-end: simulate the "happy path" usage of the `operator-sdk` binary and resulting operator project.
+- Integration: test components of the `operator-sdk` binary and features of scaffolded projects that are
+bound to external projects, such as [OLM][olm].
+- Subcommand: ensure individual subcommands function as intended with a variety of input options.
+
+## Local Test Environment
+
+If running tests locally, access to a Kubernetes cluster of server version v1.11.3 or higher is required.
+These tests require `KUBECONFIG` be set or kubeconfig file be present in a default location like `$HOME/.kube/config`.
+
+### Local clusters
+
+Two options for testing with a local cluster are [minikube][minikube] and [kind][kind].
+Ensure `KUBECONFIG` is set correctly for the chosen cluster type.
+
+## Running Tests
+
+On any PR, the entire test suite is run against your changes in a CI environment.
+Therefore it is advantageous to run all tests before pushing changes to the remote repo:
+
+```sh
+make test-sanity test-links test-unit test-subcommand test-integration test-e2e
```

Comment:
@theishshah this is the effective test suite I'd like to see under `make test` when the Makefile is simplified.

Reply:
IMO we should have a target to run all tests.

However, here we should describe all tests targets options as well. See that i might not want to test all for example. I think we need a table with all as we had before here : https://github.com/operator-framework/operator-sdk/pull/3879/files#diff-bbda1ebe0fe291e18c4cf9e65dd734eeL55-L68 and as it is done in KB. See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/CONTRIBUTING.md#what-to-do-before-submitting-a-pull-request

Reply:
E.g I want to just check the lint.

Reply:
Another point is `make test-e2e` will not work without having the k8s bin installed locally and a kind cluster. We need to describe what is required to do to run then locally.

Reply:
I assume you're talking about envtest binaries. I've added a link to the envtest setup document. You do not need a kind cluster, any cluster will do.